src/Microsoft.Framework.Runtime.Hosting/Json/JsonBuffer.cs

```diff
@@ -0,0 +1,312 @@
+            }
+            else if ((first >= '0' && first <= '9') || first == '-')
+            {
+                _token.Type = JsonTokenType.Number;
+                _token.Value = ReadNumber(first);
+            }
+            else
+            {
+                throw new JsonDeserializerException(
+                    JsonDeserializerResource.Format_IllegalCharacter(first),
+                    _token);
+            }
+
+            // JsonToken is a value type
+            return _token;
+        }
+
+        private int ReadNextChar()
+        {
+            while (true)
+            {
+                var value = _reader.Read();
+                _column++;
+
+                if (value == -1)
+                {
+                    // This is the end of file
+                    return -1;
+                }
+                else if (value == JsonConstants.LineFeed)
+                {
+                    // This is a new line. Let the next loop read the first charactor of the following line.
+                    // Set position ahead of next line
+                    _column = 0;
+                    _line++;
+
+                    continue;
+                }
+                else if (value == JsonConstants.CarriageReturn)
+                {
+                    // Skip the carriage return.
+                    // Let the next loop read the following char
+                }
+                else
+                {
+                    // Returns the normal value
+                    return value;
+                }
+            }
+        }
+
+        private string ReadNumber(int firstRead)
+        {
+            var buf = new StringBuilder();
+            buf.Append((char)firstRead);
+
+            while (true)
+            {
+                var next = _reader.Peek();
+
+                if ((next >= '0' && next <= '9') ||
+                    next == '.' ||
+                    next == 'e' ||
+                    next == 'E')
+                {
+                    buf.Append((char)ReadNextChar());
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return buf.ToString();
+        }
+
+        private void ReadLiteral(string literal)
+        {
+            for (int i = 1; i < literal.Length; ++i)
+            {
+                var next = _reader.Peek();
+                if (next != literal[i])
+                {
+                    throw new JsonDeserializerException(
+                        JsonDeserializerResource.Format_UnrecognizedLiteral(literal),
+                        _line, _column);
+                }
+                else
+                {
+                    ReadNextChar();
+                }
+            }
+
+            var tail = _reader.Peek();
+            if (tail != '}' &&
+                tail != ']' &&
+                tail != ',' &&
+                tail != -1 &&
+                tail != JsonConstants.CarriageReturn &&
```

Comment:
Can we use characters instead of these constants? It is incosistent to have chars in some places and constants in others